igv_remote/_core.py

```diff
@@ -10,28 +10,24 @@
 def _append_id(filename, id):
     return "{0}_{2}.{1}".format(*filename.rsplit('.', 1) + [id])
 
-def _parse_loc(chromosome, pos1, pos2=None, expand=20):
-    if expand < 20:
-        print("IGV expands left and right margin by at least 20bp")
-        expand=20
-    start_pos = int(pos1-expand)
-    if pos2 is None:
-        end_pos = int(pos1+expand)
-    else:
-        end_pos = int(pos2)
-    start_pos ='{:,}'.format(int(start_pos))
-    end_pos = '{:,}'.format(int(end_pos))
-    position= '{}:{}-{}'.format(chromosome, start_pos, end_pos)
-    print("Position to view: {}".format(position))
-    return position
-        
+VALID_VIEW_OPTS = ['squish', 'collapse', 'expand']
+
 class IGV_remote:
-    sock=None
-    
-    def __init__(self, view_type='collapsed', sort='base'): 
-        self.sock = None
-        self.set_viewopts(view_type=view_type, sort=sort)
-    
+
+    def __init__(self, host="127.0.0.1", port=60151, view_type='collapse', sort='base', verbose=True, recv_timeout=60):
+        self.set_viewopts(view_type,  sort)
+        self.HOST = host
+        self.PORT = port
+
+        self.verboseprint = print if verbose else lambda *a, **k: None
```

Comment:
Never thought to do this. I've used `lambda : None` plenty as a null argument for functions that expect other functions, but never in a case where the lambda itself might consume arguments. You learn something new every day!